core/src/smartcontracts/isi/mod.rs

```diff
@@ -348,7 +346,6 @@ impl<W: WorldTrait> Execute<W> for UnregisterBox {
 impl<W: WorldTrait> Execute<W> for MintBox {
     type Error = Error;
 
-    #[log]
```

Comment:
Is logging causing these problems?

Reply:
Yes, it causes deadlocks. You can checkout my branch, run trigger tests (will pass), then activate logging and rerun again (will deadlock)

Reply:
Do we have a theory for why that is?

Reply:
It's because `tracing` has a very specific idea on what `#[log]` means. I removed a whole lot of these in #2185. This also causes an ungodly amount of extra information being added with simple logging statements. 

Simply put, logging is doing a lot of extra work that you can't even see (thankfully no syscall overhead) and that's what's causing the deadlocks.